src/resdk/tables/variant.py

```diff
@@ -0,0 +1,157 @@
+""".. Ignore pydocstyle D400.
+
+=============
+VariantTables
+=============
+
+.. autoclass:: VariantTables
+    :members:
+
+"""
+import re
+from functools import lru_cache
+
+import pandas as pd
+
+from .base import BaseTables
+
+
+class VariantTables(BaseTables):
+    """A helper class to fetch collection's variant, qc and meta data.
+
+    This class enables fetching given collection's data and returning it
+    as tables which have samples in rows and variants / qc / metadata
+    in columns.
+
+    A simple example:
+
+    .. code-block:: python
+
+        # Get Collection object
+        collection = res.collection.get("collection-slug")
+
+        # Fetch collection variant data
+        tables = VariantTables(collection)
+        var = tables.variants
+
+    """
+
+    process_type = "data:mutationstable"
+    VARIANTS = "variants"
+    DEPTH = "depth"
+    COVERAGE = "coverage"
+
+    data_type_to_field_name = {
+        VARIANTS: "tsv",
+        DEPTH: "tsv",
+        COVERAGE: "tsv",
+    }
+
+    @property
+    @lru_cache()
+    def mutation_subset_genes(self) -> pd.DataFrame:
+        """Return genes for which variants are reported."""
+        return self._data[0].output["genes"]
+
+    @property
+    @lru_cache()
+    def variants(self) -> pd.DataFrame:
+        """Return variants table as a pandas DataFrame object."""
+        return self._load_fetch(self.VARIANTS)
+
+    @property
+    @lru_cache()
+    def depth(self) -> pd.DataFrame:
+        """Return depth table as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH)
+
+    @property
+    @lru_cache()
+    def coverage(self) -> pd.DataFrame:
+        """Return coverage table as a pandas DataFrame object."""
+        return self._load_fetch(self.COVERAGE)
+
+    def _download_qc(self) -> pd.DataFrame:
+        """Download sample QC data and transform into table."""
+        # No QC is given for variants data - return empty DataFrame
+        return pd.DataFrame()
+
+    @staticmethod
+    def construct_index(row) -> str:
+        """
+        Construct index of the variants table.
+
+        Index should have the form:
+        <chr>_<position>_<snp_change>_<aminoacid_pos_change>
```

Comment:
It would be great if the gene name is also added to index if we run the analysis for multiple genes at once.